src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,35 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
+        if (\count($envelope->all(OverrideSendersStamp::class)) > 0) {
+            $overrideSendersStamp = $envelope->last(OverrideSendersStamp::class);
+
+            foreach ($overrideSendersStamp->getSenders() as $senderAlias) {
+                yield from $this->getSenderFromAlias($senderAlias);
+            }
+
+            return;
+        }
+
         $seen = [];
 
         foreach (HandlersLocator::listTypes($envelope) as $type) {
             foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
                 if (!\in_array($senderAlias, $seen, true)) {
-                    if (!$this->sendersLocator->has($senderAlias)) {
-                        throw new RuntimeException(sprintf('Invalid senders configuration: sender "%s" is not in the senders locator.', $senderAlias));
-                    }
-
                     $seen[] = $senderAlias;
-                    $sender = $this->sendersLocator->get($senderAlias);
-                    yield $senderAlias => $sender;
```

Comment:
what does this provide? if this is pure refacto (as it looks to me at the moment), then this should be reverted

Reply:
To reuse the logic in above if condition, I've separated this part into another method. What is your concern, why it should be removed?

Reply:
got it thanks